addon/services/toast.ts

```diff
@@ -85,123 +93,179 @@ export default class Toast extends Service {
   }
 
-  private _handleVisibility(toast: Element, opts: ToastOptions) {
+  private _handleVisibility(toast: HTMLElement, opts: ToastOptions) {
     toast.classList.remove('toast--hidden');
     toast.classList.add('toast--visible');
 
-    this._toasts.push(toast);
+    if (opts.timeout === 'none') return;
+    this._toasts.get(toast)?.play();
+  }
 
-    if (opts.timeout === 'none') {
-      return;
-    }
+  private _destroyToast(toast: HTMLElement | null | undefined): void {
+    if (!toast) return;
 
-    later(this, () => this._destroyToast(toast), opts.timeout || DEFAULT_TOAST_TIMEOUT);
-  }
+    const destroyAnimation = this._createDestroyAnimation(toast);
+    destroyAnimation.play();
 
-  private _destroyToast(toast: Element | null): void {
-    toast?.remove();
+    destroyAnimation.finished.then(() => {
+      toast.removeEventListener('mouseenter', this._pauseCounterAnimation.bind(this));
+      toast.removeEventListener('mouseleave', this._playCounterAnimation.bind(this));
+      this._toasts.delete(toast);
+      toast?.remove();
+    });
   }
 
   private _onToastClose(evt: Event): void {
     evt.stopPropagation();
 
-    const button = (<Element>evt.target).tagName === 'I' ? (<Element>evt.target).parentElement : <Element>evt.target;
+    const button =
+      (<Element>evt.target).tagName === 'I'
+        ? (<Element>evt.target).parentElement?.parentElement
+        : (<Element>evt.target).parentElement;
 
-    if (button) {
-      this._destroyToast(button.parentElement);
-    }
+    this._destroyToast(button?.parentElement);
   }
 
-  private _setupToastEvents(toast: Element, opts: ToastOptions) {
+  private _setupToastEvents(toast: HTMLElement, opts: ToastOptions) {
     if (opts.onclick && typeof opts.onclick === 'function') {
       toast.addEventListener(
         'click',
         (e) => {
           e.stopPropagation();
-          opts.onclick && opts.onclick(e);
+          opts.onclick?.(e);
         },
         { once: true }
       );
     }
 
-    if (opts.tapToDismiss) {
-      toast.addEventListener(
-        'click',
-        () => {
-          this._destroyToast(toast);
-        },
-        { once: true }
-      );
-    }
-  }
+    this._toasts.get(toast)?.finished.then(() => {
+      this._destroyToast(toast);
+    });
 
-  private _buildCloseButton(): Element {
-    const closeButton: Element = this._buildElement(
-      'button',
-      ['toast-close-button'],
-      '<i class="fas fa-times text-size-5"></i>'
-    );
+    if (opts.timeout === 'none') return;
 
-    closeButton.classList.add('toast-close-button');
-    closeButton.innerHTML = '<i class="fas fa-times text-size-5"></i>';
-    closeButton.addEventListener('click', this._onToastClose.bind(this), { once: true });
+    toast.addEventListener('mouseenter', () => {
+      this._toasts.get(toast)?.pause();
+    });
 
-    return closeButton;
+    toast.addEventListener('mouseenter', this._pauseCounterAnimation.bind(this));
```

Comment:
Is my github drunk or are you listening for mouseenter twice ? 🤔  (line 237 & line 241 ?)

Reply:
You are not drunk, it is me here 🙃